shared/tests/utility/motion/TestKinematics.cpp

```diff
@@ -0,0 +1,173 @@
+/*
+ * This file is part of the NUbots Codebase.
+ *
+ * The NUbots Codebase is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The NUbots Codebase is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the NUbots Codebase.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright 2013 NUbots <nubots@nubots.net>
+ */
+
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+#include <catch.hpp>
+
+#include "message/input/Sensors.h"
+#include "message/motion/KinematicsModel.h"
+#include "utility/input/LimbID.h"
+#include "utility/input/ServoID.h"
+#include "utility/motion/ForwardKinematics.h"
+#include "utility/motion/InverseKinematics.h"
+
+using message::input::Sensors;
+using message::motion::BodySide;
+using message::motion::KinematicsModel;
+
+using utility::input::LimbID;
+using utility::input::ServoID;
+
+static const KinematicsModel kinematics_model = {
+    {
+        0.0,
+        0.055,
+        0.045,
+        0.2,
+        0.2,
+        0.04,
+        0.215,
+        0.13,
+        0.085,
+        0.13,
+        0.02,
+        2.0 * 0.055,
+        -1,
+        -1,
+        1,
+        1,
+        1,
+        -1,
```

Comment:
Maybe removing trailing comma to collapse this to a single line? Unless reading each value in the sequence is useful